pkg/helm/history.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2021 The Lokomotive Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helm
+
+import (
+	"sort"
+
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/release"
+)
+
+type releaseByVersionDesc []*release.Release
+
+func (rs releaseByVersionDesc) Len() int           { return len(rs) }
+func (rs releaseByVersionDesc) Less(i, j int) bool { return rs[i].Version < rs[j].Version }
+func (rs releaseByVersionDesc) Swap(i, j int)      { rs[i], rs[j] = rs[j], rs[i] }
+
+func sortAndTrimHistory(h []*release.Release, max int) []*release.Release {
+	sort.Sort(sort.Reverse(releaseByVersionDesc(h)))
+
+	if max > len(h) {
+		max = len(h)
+	}
+
+	return h[:max]
+}
+
+// GetHistory returns at most max elements of the Helm history in descending
+// version order, that is, the first element returned is the newest version of
+// the release.
+func GetHistory(client *action.History, name string, max int) ([]*release.Release, error) {
```

Comment:
Maybe make `*action.History` an interface with `Run(name string) ([]*release.Release, error)` method, so we can have blackbox tests for it using self-written mock client?